google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -109,19 +109,24 @@ public int read(ByteBuffer dst) throws IOException {
           amt = channel.read(dst);
           break;
         } catch (StorageException exs) {
-          if (exs.getMessage().contains("Connection closed prematurely") && reopens < maxChannelReopens) {
-            // this error isn't marked as retryable since the channel is closed;
+          if (reopens < maxChannelReopens && (
+              exs.getMessage().contains("Connection closed prematurely")
+          || exs.getCause() instanceof javax.net.ssl.SSLHandshakeException
```

Comment:
Should we trap the parent class exception `javax.net.ssl.SSLException` here instead of the more specific `javax.net.ssl.SSLHandshakeException`? I think we may have seen the more broad `javax.net.ssl.SSLException` in a few stack traces.